eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -31,6 +31,7 @@ steps:
         | "$(VcpkgCommit)"
         | $(Agent.Os)
         | $(${{ parameters.DependenciesVariableName }})
+        | $(VCPKG_DEFAULT_TRIPLET)
```

Comment:
@weshaggard, @danieljurek - We are adding the triplet name to the restore key of the cache to make it more unique and force a cache miss.

This was causing issues because we temporarily changed the triplet value to solve a different issue (and reverted it back), but since it is still within the 7 day window, the cache had the incorrect packages stored corresponding to the wrong triplet (x86-windows-static**-md** instead of x86-windows-static).

**For context:**
- The temporary change made - https://github.com/Azure/azure-sdk-for-c/pull/1338
- Which we reverted back to what it originally was here - https://github.com/Azure/azure-sdk-for-c/pull/1394

https://docs.microsoft.com/en-us/azure/devops/pipelines/release/caching?view=azure-devops#can-i-clear-a-cache
> Clearing a cache is currently not supported. However you can add a string literal (e.g. version2) to your existing cache key to change the key in a way that avoids any hits on existing caches.

Reply:
FYI, @vhvb1989, @antkmsft, @RickWinter, if this issue comes up again.

Once we have a new release of Vcpkg, we can leverage their own binary caching feature which might be less error-prone and flexible, moving forward: https://github.com/microsoft/vcpkg/pull/11204

Reply:
@ahsonkhan please comment the vcpkg.yml file with these details.  Years from now it will be helpful to know why that triplet is there and I doubt anyone will remember this issue

Reply:
@RickWinter 2 months later, I remembered ahahaha..  Happened again here: https://github.com/Azure/azure-sdk-for-cpp/pull/1042

Reply:
> please comment the vcpkg.yml file with these details. 

https://github.com/Azure/azure-sdk-for-c/issues/1534